### DISCUSSION TITLE
更新のお知らせ

### DISCUSSION THREAD

**Discussion by dounokouno** [dounokouno/TransmitMail#26] [General]
更新のお知らせを書いていきます。

# 2020年12月28日追記

GitHub で Discussions 機能が使えるようになったので、以前の issue から discussion に convert しました。

**Comment by dounokouno** [comment]
入力必須項目に  `0` を入力した場合に入力必須チェックに引っかかってしまう件を修正した v1.5.11 をリリースしました。

# 修正内容

https://github.com/dounokouno/TransmitMail/commit/ff71cb36e8380222a7e5f3a6c19253b10fc07518

**Comment by dounokouno** [comment]
`x` `y` を含む項目名が利用できない件を修正した v1.5.12 をリリースしました。

# 経緯

送信ボタンに `<input type="image">` を利用した場合に `<input type="image">` のクリックされた座標を送信するブラウザの仕様により、TransmitMail のループ出力を利用した際に `x` `y` の値が意図せず送信、表示されてしまう、といった問題がありました。

この `x` `y` の出力を抑制するために「無視する項目名」に `x` `y` を追加したところ、「 `x` `y` を含む項目名」が無視されるようになってしまい、例えば「Compay」のように `y` を含む項目（またはその他の `x` `y` を含む項目）が無視されてしまう状態になっていた。（該当 commit -> https://github.com/dounokouno/TransmitMail/commit/551bce60b60f2c730b70102a164391dc4e47c76b ）

そこで、「無視する項目名」を完全一致で判別するように修正し、`x` `y` を含む項目名を利用できるようになった。

# 修正内容
- https://github.com/dounokouno/TransmitMail/commit/0273b6279d761f1fe663bdc3a962b7ff23230a48  
  入力情報として除外する項目の判別処理を修正
- https://github.com/dounokouno/TransmitMail/commit/33bc5de20a37850ca7c9e0b77aff6c9b5542a410  
  input.html に x、y を含む項目名のフィールドを追加

# この件に関する過去の質問

https://github.com/dounokouno/TransmitMail/issues/2#issuecomment-43404299

> confirm.html
> で初めに確認したところ、
> x = 54
> y = 11
> というのが出てしまったので、confirm_kobetsu.html　を使用することにしました。
> すると「お問い合わせありがとうございます」メールにも
> x = 54
> y = 11
> というのが入ってしまっております。
> これを削除したいのですが、どのようにしたらよろしいのでしょうか？

**Comment by dounokouno** [comment]
自動返信メールに CC と BCC の追加と、変数名などを修正した v2.3.0 をリリースしました。

元々は自動返信メールに CC と BCC の追加いただいた PR （ https://github.com/dounokouno/TransmitMail/pull/52 ）をマージするだけの予定だったのですが、自動返信メールの CC 、 BCC の設定値との兼ね合いで、下記の設定名を変更しました。

- `to_email` → `email`
- `to_subject` → `subject`
- `auto_reply_email` → `auto_reply_email_input_name`

ので、 v2.2 以下からアップデートする場合は、 lib/TransmitMail.php のアップデートとあわせて上記の設定値を修正する必要があります。 v2.2 以下からアップデートされる場合はご注意ください。

例えば、 config/config.yml に下記のように書いていた場合、

```yml
to_email: info@example.com
```

下記のように修正が必要です。

```yml
email: info@example.com
```

**Comment by dounokouno** [comment]
いくつかの修正を行った v2.4.0 をリリースしました。

# 主な修正内容

- init メソッドを追加し、 init メソッドで各種初期設定を行うように修正しました
- ファイルの表示時には session を利用しないように修正しました（ファイルの表示時に session を利用してしまうことで、 CSRF トークンが再生成されてしまう場合があるため）

# init メソッドの追加により、 index.php が一部変更になりました

init メソッドの追加により、 index.php の内容が一部変更になりました。

具体的には、 v2.4.0 からは index.php は下記のように記述します。

```php
require_once 'lib/TransmitMail.php';
$tm = new TransmitMail();
$tm->init('config/config.yml');
$tm->run();
```

詳細は下記のコミットを参照ください。

6342be0

**Comment by dounokouno** [comment]
テンプレートファイルから別のテンプレートファイルを呼び出す include 機能を実装した v2.5.0 をリリースしました。

インクルードの記法は Wiki ページを参照ください。

https://github.com/dounokouno/TransmitMail/wiki/テンプレートの記述方法#インクルード

**Comment by dounokouno** [comment]
[2020年8月11日にリリースされた WordPress 5.5](https://ja.wordpress.org/2020/08/11/eckstine/) 上で TransmitMail がうまく動作しなくなったので、対応した [v2.6.0](https://github.com/dounokouno/TransmitMail/releases/tag/2.6.0) をリリースしました。

細かい修正が主だったのですが、一部に breaking changes が含まれる（ `lib` ディレクトリ以下を最新のファイルに入れ替えただけではアップグレードできない）ので、 v2.5.0 → v2.6.0 としました。

v2.5.0 → v2.6.0 の変更点は下記を参照ください。

https://github.com/dounokouno/TransmitMail/compare/2.5.0...2.6.0

v2.5.0 から v2.6.0 にアップデートする場合は、下記の2点の対応を行なってください。

- 下記の3つのファイルを最新のファイルに置き換える
    - `lib/TransmitMail.php`
    - `lib/qdmail.php`
    - `lib/qdsmtp.php`
- 入力確認画面のテンプレート（初期値では `confirm.html` ）の `<input type="hidden" name="page" value="...">` を `<input type="hidden" name="page_name" value="...">` に修正する

v2.6.0 では標準のテンプレートファイルの `form` 要素から `action` 属性を削除していますが、この部分は好みで対応してください。

# `page` を `page_name` に変更した理由の補足説明

WordPress 内で利用している変数名と被っていたことが原因でした。 WordPress v5.4 系では問題なく動作していたので、 v5.5 から WordPress 内で利用している変数名をおかしなところで利用することが許容されなくなったようです。

- 参考: [フォーム作成時、inputのnameに予約語を使うと404になる – PlanFollow-up Company](https://plan-follow-up.com/2015/10/27/query-variables/)

**Comment by dounokouno** [comment]
TransmitMail の v2.5.0 〜 v2.6.1 で脆弱性が発見されたため、修正版の [v2.6.2](https://github.com/dounokouno/TransmitMail/releases/tag/2.6.2) をリリースしました。

脆弱性の内容と対応方法に関しましては、下記ブログ記事にて公開しております。該当バージョンをご利用の方はご確認、ご対応いただきますよう、お願いをいたします。

https://dounokouno.com/2022/01/25/about-the-vulnerability-of-transmitmail-v2-5-0-v2-6-1/

**Reply by Randing** [reply]
長く、愛用させていただいております。いつもありがとうございます。

ソースコードによるバージョン確認を行おうと思ったのですが、
下記のファイルが見当たりませんでした。
./lib/common.php

チェックモードにて、無事確認できたのですが（v2.6.1）
利用中のバージョンによっては上記のファイルは存在しないでしょうか。
ご連絡までにて、よろしくお願い申し上げます。

**Reply by dounokouno** [reply]
上記件に関しまして、確認しましたところ、記事内のファイル名が `./lib/common.php` となっておりましたが、正しくは `./lib/TransmitMail.php` でした。（その下のリンク先は合っていました。）

早速ブログ記事を修正いたしました。

https://dounokouno.com/2022/01/25/about-the-vulnerability-of-transmitmail-v2-5-0-v2-6-1/

ご指摘いただきありがとうございました。

**Reply by Randing** [reply]
迅速にご確認いただき有難うございました！
